RPIPs/RPIP-30.md

```diff
@@ -47,45 +47,77 @@ the protocol goal of attracting minipool creation (and rETH supplying).
-    `proposed_method_share`, and how to combine them
-    - The tree specs SHALL follow existing processes for release and modification
-- The next significant smart contract upgrades SHALL update the withdrawal process
-  - Withdrawals SHALL be a 2-step process
-    - A user MAY set some amount of RPL to "withdrawing"
-    - Once RPL has been in the withdrawing state for 28 days, it MAY be withdrawn from the protocol
-      - All "withdrawing" RPL MUST be withdrawn at one time
-      - The 28-day value SHOULD be a pDAO-controlled setting that starts at 28 days
-    - There SHALL only be a single time stored for "withdrawing" RPL; this means that if a user sets
-      additional RPL to withdrawing after some RPL is already "withdrawing", they must wait for 28
-      days after the later action
-    - "Withdrawing" RPL SHALL NOT be eligible for RPL rewards or voting power
-  - If this smart contract upgrade has not yet happened, the pDAO SHALL update the withdrawal
-    threshold as follows:
-    - Within the two weeks after X=3 rewards go out, set `node.per.minipool.stake.maximum` to 1.0
-    - Within the two weeks after X=6 rewards go out, set `node.per.minipool.stake.maximum` to 0.6
-- Except for not including "withdrawing" RPL, there SHALL NOT be any changes to voting power based
-  on this RPIP
-
-## Additional Considerations
+### Final states
+- For periods X>5, a node's share of reward SHALL be `proposed_method_share`
+- The rewards tree specs SHALL follow pre-existing processes for release and modification
+- The next significant smart contract upgrades SHALL update the RPL withdrawal process to be a
+  2-step process
+  - A user MAY set any amount of "Staked" RPL beyond the value of 15% of the node's borrowed ETH to
+    "Unstaking"
+  - Once RPL has been "Unstaking" for `unstaking_period` days, it can be withdrawn
+    - All RPL that was "Unstaking" MUST be withdrawn at one time
+    - `unstaking_period` SHOULD be a pDAO-controlled setting initialized to 28 days
+  - There SHALL only be a single time stored for "Unstaking" RPL; this means that if a user sets
+    additional RPL to "Unstaking" after some RPL is already "Unstaking", they must wait for
+    `unstaking_period` before any is available to withdraw
+    - All RPL that was already "Unstaking" for `unstaking_period` SHOULD be withdrawn when a user
+      sets additional RPL to "Unstaking"
+- Voting power calculations SHALL NOT be changed based on this RPIP (though do note that they SHALL
+  NOT count "Unstaking" RPL)
+
+### Transitioning towards final states
+- For periods X=1 to X=5: a node's share of rewards is `(X/6)*proposed_method_share + ((6-x)/6)*current_method_share`
+  - For these periods, the rewards tree spec MUST specify how to calculate `current_method_share`,
+    `proposed_method_share`, and how to combine them to get a node's share of rewards
+  - The rewards tree specs SHALL follow pre-existing processes for release and modification
```

Comment:
That process can't really fail -- if it fails to go through the easy way, it goes to vote.

https://dao.rocketpool.net/t/meta-proposal-for-a-watchtower-and-rewards-tree-change-process/1267
> The changes would be first presented in a post here and relayed in our Discord to allow for a proper discussion. If contentious, they should be run through our usual Snapshot voting process.

--

For the forward looking thing, since it's a future vote it could clearly eliminate/replace this bullet. Eg, could become "SHALL follow the processes for release and modification in RPIP-blah". Could even entirely supersede this RPIP with a new one including that one change if we believe that's cleanest.

Reply:
The vote can fail though. And you're in the situation where you've ratified a thing that says 'x will happen' and x can't happen because when voted on specifically, it was voted down.

Reply:
Arguably it should just be 'this RPIP if ratified authorizes changes to the smartnode to achieve these goals? Or something.

Reply:
I think I might just delete the lines -- in the spirit of your earlier comment about not specifying stuff that's not changed. If we don't say we change the process, we don't change the process. Unlike voting, I don't think this is likely to be a common question.